### PR TITLE
Issue #12055 Support reversed menu animation when anchor is on the left.

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by rocketsroger** [comment]
https://user-images.githubusercontent.com/6579541/165130264-215ee8e4-adf5-4f0c-bd37-e0b3a18b5c79.mp4

File: components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt [hunk 1]
```diff
@@ -23,24 +23,27 @@ internal fun PopupWindow.displayPopup(
     containerView.measure(spec, spec)
 
     val (availableHeightToTop, availableHeightToBottom) = getMaxAvailableHeightToTopAndBottom(anchor)
+    val (availableWidthToLeft, availableWidthToRight) = getMaxAvailableWidthToLeftAndRight(anchor)
     val containerHeight = containerView.measuredHeight
 
     val fitsUp = availableHeightToTop >= containerHeight
     val fitsDown = availableHeightToBottom >= containerHeight
+    val reversed = availableWidthToLeft < availableWidthToRight
```
Comment (Amejia481) [inline components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt:31]: Could we have issues with right to left locales?

**Review by Amejia481 — APPROVED** [review]
The changes looks good in general, just [one question](https://github.com/mozilla-mobile/android-components/pull/12054#discussion_r857814259) as I'm not sure if could have issues on right to left locales. Not blocking as I'm not really sure if it's an issue.

File: components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt (hunk 1 above)
Comment (rocketsroger) [reply components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt:31]: From my testing there's no issue.  When we detect right to left, we use the opposite (mirroring) animation.

Comment (Amejia481) [reply components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt:31]: 👍🏽

Comment (rocketsroger) [reply components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt:31]: I'll add some RTL tests.  Thanks

**Comment by rocketsroger** [comment]
> The changes looks good in general, just [one question](https://github.com/mozilla-mobile/android-components/pull/12054#discussion_r857814259) as I'm not sure if could have issues on right to left locales. Not blocking as I'm not really sure if it's an issue.

Thanks, added some RTL tests.

**Pull request merged**